src/middlewared/middlewared/plugins/vm/vms.py

```diff
@@ -377,6 +380,24 @@ async def do_update(self, id_, data):
         vm_data = await self.get_instance(id_)
         if new['name'] != old['name']:
             await self.middleware.run_in_thread(self._rename_domain, old, vm_data)
+            try:
```

Comment:
I don't like creating 2 separate threads for small operations. Either convert the entirety of this method to being synchronous or add another method that does all blocking IO so that we can do it in 1 thread total